adam_core/propagator/propagator.py

```diff
@@ -73,24 +73,14 @@ def propagate_orbits(
             Propagated orbits.
         """
         if num_jobs is None or num_jobs > 1:
-            orbits_split = [
-                orbit_chunk for orbit_chunk in _iterate_chunks(orbits, chunk_size)
-            ]
-
-            p = mp.Pool(
-                processes=num_jobs,
-            )
-
-            propagated_list = p.starmap(
-                propagation_worker,
-                zip(
-                    orbits_split,
-                    repeat(times),
-                    repeat(self),
-                ),
-            )
-            p.close()
-            p.join()
+            with concurrent.futures.ProcessPoolExecutor(max_workers=num_jobs) as executor:
```

Comment:
This doesn't really have to do with this PR, but this might be a good opportunity to rename `num_jobs` to `workers` or `processes` or `parallelism` or something equivalent, since a job is a unit of work and the parameter is really about how many separate processes to use.

Reply:
I like this suggestion a lot, I'll do it here.